### DISCUSSION TITLE
Transcode buffer sizes: no effect at all?

### DISCUSSION THREAD

**Discussion by udippel** [gerbera/gerbera#2997] [General]
Based on earlier experiments, i have finally started to play with these numbers, that are given as "Required"., None of my experiments with the values have brought any difference at all encoding with mpeg2video. No effects on quality, no effect on system load, no effect on startup
`<buffer size="32768" chunk-size="4096" fill-size="0"/>
`works exactly identical to `<buffer size="16777216" chunk-size="131072" fill-size="262144"/>
`with ffmpeg here, on video signals. 

Can anyone confirm this, or refute this? Could it be that these values are just not read and not considered by ffmpeg? 

At least, this could explain the troubles of encoding into MP4, e.g. #2858

**Comment by whyman** [comment]
Those buffers are in gerbera, used to read/write connected to the ffmpeg pipes.

**Comment by udippel** [comment]
Understood. Since I am running a low-resources box, I wanted to optimize the sizes for lowest resource usage. Thereby I noticed that whatever values I entered, there was no observable difference w.r.t. the items mentioned above. Therefore my question as above. 
In case these values don't matter at all, or are overruled by ffmpeg, we might take out the 'required' tag.

**Comment by KarlStraussberger** [comment]
As said they effect the amount of memory gerbera acquires to do the transcoding. That's why they are required. The do not change the transcoding or are passed to any external program.
